src/flashops.c

```diff
@@ -72,11 +81,60 @@ int8_t writeDataflash(MSHW* ms, unsigned int translated_addr, uint8_t val)
 	return modified;
 }
 
+/* Generate and set a random serial number valid for Mailstation
+ *
+ * Disassembly of codeflash found that the verification process for serial num
+ * is that it starts at 0x7FFC8, is 16 bytes long, each character must be
+ * ASCII '0'-'9', 'A'-'Z', 'a'-'z', or '-'
+ * In all observed Mailstations, the last character is '-', but it does
+ * not seem to be enforced in Mailstation firmware. Nevertheless, the last char
+ * is set to a '-'
+ */
+void set_rnd_ms_ser(uint8_t *buf)
```

Comment:
Also: since this is DF specific, maybe call it `set_df_serial_random` so it's clear this is only meant for dataflash.

Reply:
Refactored, made this function accept ms_ctx and modify the dataflash from there. All self contained now.

And moved all DF related funcs to prefix with "df_"